### PR TITLE
Feature:  Draw arrow markers on links.

### LINKED ISSUE #127: Link arrow
## Description

**As a** user,

**I want** to see an arrow on link,

**so that** I can understand where come from the link

### PR DESCRIPTION
**Pull request by Hundraw**
Close #127

### CODE REVIEW

File: src/draw/DefaultDrawer.js [hunk 1]
```diff
@@ -122,6 +122,9 @@ class DefaultDrawer {
         state: false,
         target: null,
       },
+      zoom: {
```
Comment (Zorin95670) [inline src/draw/DefaultDrawer.js:125]: documentation ?

File: src/models/DefaultData.js [hunk 2]
```diff
@@ -233,6 +233,29 @@ class DefaultData {
       }, []);
   }
 
+  /**
+   * Get all used link definitions.
+   *
+   * @returns {ComponentLinkDefinition[]} - List of link definitions.
+   */
+  getUsedLinkDefinitions() {
```
Comment (Zorin95670) [inline src/models/DefaultData.js:241]: rework on documentation and code

File: src/models/ComponentLinkDefinition.js [hunk 3]
```diff
@@ -13,6 +13,14 @@ class ComponentLinkDefinition {
    * @param {string} [props.color='black'] - Color of the link.
    * @param {number} [props.width=2] - Width of the link.
    * @param {number[]} [props.dashStyle] - Dash style of the link. See stroke-dasharray of svg.
+   * @param {object} [props.marker] - Marker of the link.
+   * @param {string} [props.marker.viewBox='0 0 5 5'] - Viewbox of the marker.
+   * @param {number} [props.marker.width=5] - Width of the marker.
+   * @param {number} [props.marker.height=5] - Height of the marker.
+   * @param {number} [props.marker.refX=4] - X reference of the marker.
+   * @param {number} [props.marker.refY=2.5] - Y reference of the marker.
+   * @param {string} [props.marker.orient='auto'] - Orientation of the marker.
```
Comment (Zorin95670) [inline src/models/ComponentLinkDefinition.js:22]: list possible orient

File: changelog.md [hunk 4]
```diff
@@ -5,6 +5,22 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html)
 
+## [Unreleased]
+
+### Added
+
+  - Add `getUsedLinkDefinition` in `DefaultData.js` to get the link definitions used by components.
+  - Add `__initializeArrowMarker` method in `DefaultDrawer.js`.
+  - Add `actions.zoom.scale` in constructor of `DefaultDrawer.js` to get the zoom scale.
```
Comment (Zorin95670) [inline changelog.md:14]: missing marker definition

**Review by Zorin95670 — CHANGES_REQUESTED** [review]

File: src/draw/DefaultDrawer.js [hunk 5]
```diff
@@ -799,6 +804,34 @@ class DefaultDrawer {
...
+    const definitions = this.pluginData.getUsedLinkDefinitions();
+    const arrows = this.svg.select('defs').selectAll('arrow')
+      .data(
+        definitions,
+        (data) => `${data.attributeRef}-${data.sourceRef}-${data.targetRef}`,
+      )
+      .join('marker')
+      .attr('class', 'arrow');
+
+    arrows
```
Comment (Zorin95670) [inline src/draw/DefaultDrawer.js:822]: Reverse link missing

File: src/draw/DefaultDrawer.js (hunk 1 above)
Comment (Hundraw) [reply src/draw/DefaultDrawer.js:125]: we have never documented the value of the actions

File: src/models/DefaultData.js (hunk 2 above)
Comment (Hundraw) [reply src/models/DefaultData.js:241]: done

File: src/draw/DefaultDrawer.js (hunk 5 above)
Comment (Hundraw) [reply src/draw/DefaultDrawer.js:822]: done

File: changelog.md (hunk 4 above)
Comment (Hundraw) [reply changelog.md:14]: done

File: src/models/ComponentLinkDefinition.js (hunk 3 above)
Comment (Hundraw) [reply src/models/ComponentLinkDefinition.js:22]: done

**Review by Zorin95670 — APPROVED** [review]

**Pull request merged**